### PR TITLE
term click sounds rework

### PR DESCRIPTION
**Pull request by SubAt0m1c**
nashes hasnt complained about this version yet so it should be good

The code might be terrible, idk i have a really bad feeling but i looked through it and dont really know what id change

### CODE REVIEW

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalSounds.kt [hunk 1]
```diff
@@ -41,35 +43,66 @@ object TerminalSounds : Module(
...
+        onMessage("The gate has been destroyed!", false, { enabled && shouldReplaceSounds }) { mc.thePlayer.playSound("note.pling", 8f, 4f) }
+
+        onMessage("The Core entrance is opening!", false, { enabled && shouldReplaceSounds }) { mc.thePlayer.playSound("note.pling", 8f, 4f) }
+    }
+
+    private fun clickSlot(slot: Int) {
+        if (
+            (currentTerm != TerminalTypes.ORDER && slot !in TerminalSolver.solution) ||
+            (currentTerm == TerminalTypes.ORDER && slot != TerminalSolver.solution.first()) ||
+            (currentTerm == TerminalTypes.MELODY && slot !in arrayOf(43, 34, 25, 16))
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalSounds.kt:85]: uh make the (currentTerm == TerminalTypes.MELODY && slot == 43) check here instead?

**Review by freebonsai — CHANGES_REQUESTED** [review]
change the thingy i commented

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalSounds.kt (hunk 1 above)
Comment (SubAt0m1c) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalSounds.kt:85]: unless im misunderstanding, the sound should be playing on every melody slot. the slot == 43 is only to check if its the last slot to play the completion sound.

**Comment by SubAt0m1c** [comment]
nevermind melody doesnt work i gotta fix it

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalSounds.kt (hunk 1 above)
Comment (freebonsai) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalSounds.kt:86]: aha ok

**Review by freebonsai — APPROVED** [review]

**Pull request merged**